tests/tests.cpp

```diff
@@ -98,17 +98,54 @@ TEST_CASE("test if bad data is inputted", "[weight=1][part=1]")
 //     REQUIRE(true);
 // }
 
-// // Advanced test cases
-// TEST_CASE("testTraverse1", "[weight=1][part=2]")
-// {
-//     // testing traversing the tree for the shortest path (dataset 1)
-//     REQUIRE(true);
-// }
-// TEST_CASE("testTraverse2", "[weight=1][part=2]")
-// {
-//     // testing traversing the tree for the shortest path (dataset 2)
-//     REQUIRE(true);
-// }
+// Advanced test cases
+TEST_CASE("testTraverse1", "[weight=1][part=2]")
+{
+    // testing traversing the tree for the shortest path (dataset smallBasic)
+    Graph *graph = new Graph("datasets/smallBasic.csv");
+    std::vector<Node*> path = graph->BFS(graph->getNodes()->at(0)); // first node is 1
+
+    std::string ss = "";
+    for (unsigned i = 0; i < path.size(); i++) {
+        ss += path.at(i)->getZone() + " ";
+    }
+    REQUIRE(ss == "1 2 3 ");
```

Comment:
why can't you just compare vectors and see if the vectors are the same

Reply:
idk if there's an easy == for vectors